File: data/multilang_comparison/program_c/src/func24.c
#include "program_c.h" 
array_t* func24(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array68 = func25(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array68->id);
      free(params0.data);
      array68->refC--;
      if(array68->refC == 0) {
         free(array68->data);
         free(array68);
         DEBUG_FREE(array68->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array77 = func26(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array77->id);
      free(params0.data);
      array77->refC--;
      if(array77->refC == 0) {
         free(array77->data);
         free(array77);
         DEBUG_FREE(array77->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array82 = func27(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array82->id);
   free(params0.data);
   return array82;
}

File: data/multilang_comparison/program_nim/src/func24.nim
proc func24*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array163: ptr ArrayT = func25(addr params0, getPath(), loopsFactor)
      debugReturn(array163.id)
      dealloc(params0.data)
      dec array163.refC
      if array163.refC == 0:
         dealloc(array163.data)
         dealloc(array163)
         debugFree(array163.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array173: ptr ArrayT = func26(addr params0, getPath(), loopsFactor)
      debugReturn(array173.id)
      dealloc(params0.data)
      dec array173.refC
      if array173.refC == 0:
         dealloc(array173.data)
         dealloc(array173)
         debugFree(array173.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array178: ptr ArrayT = func27(addr params0, getPath(), loopsFactor)
   debugReturn(array178.id)
   dealloc(params0.data)
   return array178

